src/HttpHeader.cc

```diff
@@ -1128,14 +1128,6 @@ HttpHeader::putSc(HttpHdrSc *sc)
     mb.clean();
 }
 
-void
-HttpHeader::putWarning(const int code, const char *const text)
```

Comment:
> PR description: Stop adding new Warning headers. Existing Warning are removed or retained per RFC 7234.

Can we go all the way (like the PR version before the forced push attempted to do IIRC) and just stop treating _all_ Warnings specially? Dropping all special Warning code would be great, from code maintenance point of view and possibly even from performance point of view. If the latest specs allow us to do that, why not to do that? Please do not misinterpret my question as a disagreement with your decision to go half-way. At this point, I am only asking for the reason behind this decision without assessing the decision itself. Thank you.

Reply:
I would like it to go further on the code removal. But am not able to test the logic changes at this time until at least PR #694 is merged. For now, this is the minimal change for compliance.

Reply:
FWIW, I think it would be much better to remove all (special handling of) Warnings (if that is possible) from Squid in one/this comprehensive PR than to rush this PR (that nobody urgently needs) with partial changes while increasing overheads and the probability that the critical Warning-mangling code will remain in Squid for a long time.

Briefly disclosing (in the PR description) why this PR did not go all the way would be useful IMO.

I do not insist on that or any other changes though.

Reply:
I believe this is now done as of [39f33fb](https://github.com/squid-cache/squid/pull/1072/commits/39f33fb643b1d903e18cf630b033fbf492226dee).
The semaphoreci tests need to be updated before this new behaviour will pass pre-merge checks.

Reply:
Thank you. Resolving.